blocks/api/serializer.js

```diff
@@ -51,24 +51,23 @@ export function getSaveContent( blockType, attributes ) {
 	}
 
 	// Adding a generic classname
-	const addAdvancedAttributes = ( element ) => {
+	const addGeneratedClassname = ( element ) => {
```

Comment:
I think serializer should have no knowledge of class name after these changes, given that className behavior is considered to be an extension of a core behavior where we simply allow hooks to override props. Really it seems the only thing we still rely on here is to call `classnames`, but this could be moved into the block hook logic itself.

__Edit:__ Nevermind, I see from the original comment that we are still accounting for the default class name. However, where do we use this feature? Can we drop it? I don't think we have use for it. Why couldn't a block author just assign `className` in `save` than by a separate block type property? Fine to do separately.

Reply:
> However, where do we use this feature? Can we drop it? 

Yes, we do use it, that's how the generated className for all blocks is added.

> Why couldn't a block author just assign className in save than by a separate block type property? 

I was also thinking it could be a separate extension. But yeah, let's fix this separately